hilti/toolchain/include/ast/attribute.h

```diff
@@ -136,7 +136,12 @@ class AttributeSet : public Node {
     bool has(std::string_view tag) const { return find(tag) != nullptr; }
 
     /** Adds an attribute to the set. */
-    void add(ASTContext* ctx, Attribute* a) { addChild(ctx, a); }
+    void add(ASTContext* ctx, Attribute* a) {
+        addChild(ctx, a);
+        // Combine this location with the attribute's location so this spans the range
+        auto new_meta = Meta(meta().location().mergedRange(a->location()), meta().comments());
+        setMeta(std::move(new_meta));
+    }
```

Comment:
How about one more intermediary: add `mergeLocation()` to `Meta` so that this can become:

```suggestion
    void add(ASTContext* ctx, Attribute* a) {
        addChild(ctx, a);
        // Combine this location with the attribute's location so this spans the range
        setMeta(meta()->mergeLocation(a->location()));
    }
```

Then it's easier to reuse elsewhere if other similar cases come up.